examples/batch/main.cc

```diff
@@ -0,0 +1,101 @@
+#include "opentelemetry/sdk/trace/tracer_provider.h"
+#include "opentelemetry/trace/provider.h"
+// Using an exporter that simply dumps span data to stdout.
+#include "opentelemetry/exporters/ostream/span_exporter.h"
+#include "opentelemetry/sdk/trace/batch_span_processor.h"
+
+#include <chrono>
+#include <thread>
+
+constexpr int kNumSpans = 10;
+
+namespace
+{
+
+void initTracer()
+{
+  /*                         SET UP EXPORTER                                    */
+  auto exporter = std::unique_ptr<sdktrace::SpanExporter>(
+      new opentelemetry::exporter::trace::OStreamSpanExporter);
+
+  /*              CONFIGURE BATCH SPAN PROCESSOR PARAMETERS                     */
+
+  // We make the queue size `KNumSpans`*2+5 because when the queue is half full, a preemptive notif
+  // is sent to start an export call, which we want to avoid in this simple example.
+  const size_t max_queue_size = kNumSpans * 2 + 5;
+
+  // Time interval (in ms) between two consecutive exports.
+  const std::chrono::milliseconds schedule_delay_millis = std::chrono::milliseconds(3000);
+
+  // We export `kNumSpans` after every `schedule_delay_millis` milliseconds.
+  const size_t max_export_batch_size = kNumSpans;
+
+  auto processor = std::shared_ptr<sdktrace::SpanProcessor>(new sdktrace::BatchSpanProcessor(
+      std::move(exporter), max_queue_size, schedule_delay_millis, max_export_batch_size));
+
+  // Wait for processor's worker thread to initialize.
+  std::this_thread::sleep_for(std::chrono::milliseconds(50));
```

Comment:
Why do we need this? Could we go without this?

Reply:
Oops. Yes, we can. Now that we aren't checking `if(buffer_.size() < max_export_batch_size_` in the background routine of the batch processor, we can get rid of this.

Earlier, by the time this condition was checked in the worker thread, we already had `max_export_batch_size_` spans in the buffer, meaning that we didn't honor the `schedule_delay_millis_` time interval. Also, we'd not wait 3 seconds after printing out `Exporting first batch of 10 spans in 3 seconds ...\n` and would instead directly export span data to stdout, not giving the user a chance to see what's going on. Thus, I added the 50ms sleep.

Thank you for the heads up!